### PR TITLE
Batch upload

### PR DESCRIPTION
**Pull request by JeffreyThiessen**

### CODE REVIEW

**Comment by JeffreyThiessen** [comment]
resolves #6

File: progress/upload_status.py [hunk 1]
```diff
@@ -38,6 +39,16 @@ def get_directory_status(directory, required_file_list):
         return result
 
     file_list = next(os.walk(directory))[2]  # Gets the list of files in the directory
+
+    # Legacy upload catch
+    # When the irida-miseq-uploader (old uploader) ran it generated a .miseqUploaderComplete file
+    # To prevent uploading runs that used this old system, we assume runs with this file are COMPLETE
+    # They can still be uploaded using the --force option, but will not be picked up automatically with --batch
```
Comment (deepsidhu85) [inline progress/upload_status.py]: Reword this to be more specific

File: progress/upload_status.py [hunk 2]
```diff
@@ -78,14 +93,19 @@ def write_directory_status(directory_status, run_id=None):
     if not os.access(directory_status.directory, os.W_OK):  # Cannot access upload directory
         raise exceptions.DirectoryError("Cannot access directory", directory_status.directory)
 
+    # if directory_status.is_invalid():
```
Comment (deepsidhu85) [inline progress/upload_status.py]: Remove this

File: tests_integration/test_end_to_end.py [hunk 3]
```diff
@@ -171,6 +174,95 @@ def test_valid_miseq_upload(self):
...
+        upload_result = batch_upload_single_entry(path.join(path_to_module, "fake_batch_data"))
+
+        # Make sure the upload was a success
+        self.assertEqual(upload_result, 0)
+
+        # Verify the files were uploaded
+        sample_list = test_api.get_samples(project_id)
+
+        sample_1_found = False
+        sample_2_found = False
```
Comment (deepsidhu85) [inline tests_integration/test_end_to_end.py]: Change to sample_2_not_found = True

File: tests_integration/test_end_to_end.py [hunk 4]
```diff
@@ -171,6 +174,95 @@ def test_valid_miseq_upload(self):
...
+                        sequence_files[1]['fileName']
+                    ]
+                    expected_sequence_file_names = [
+                        'batch01-1111_S1_L001_R1_001.fastq.gz',
+                        'batch01-1111_S1_L001_R2_001.fastq.gz'
+                    ]
+                    self.assertEqual(res_sequence_file_names.sort(), expected_sequence_file_names.sort())
+                elif sample.sample_name == "batch02-2222":
+                    # this one should not be found
+                    sample_2_found = True
```
Comment (deepsidhu85) [inline tests_integration/test_end_to_end.py]: Change this to sample_2_not_found = False

File: tests_integration/test_end_to_end.py [hunk 5]
```diff
@@ -171,6 +174,95 @@ def test_valid_miseq_upload(self):
...
+                        sequence_files[1]['fileName']
+                    ]
+                    expected_sequence_file_names = [
+                        'batch03-3333_S1_L001_R1_001.fastq.gz',
+                        'batch03-3333_S1_L001_R2_001.fastq.gz'
+                    ]
+                    self.assertEqual(res_sequence_file_names.sort(), expected_sequence_file_names.sort())
+
+        self.assertEqual(sample_1_found, True)
+        self.assertEqual(sample_2_found, False)
```
Comment (deepsidhu85) [inline tests_integration/test_end_to_end.py]: Change this to self.assertEqual(sample_2_not_found, True)

File: progress/upload_status.py (hunk 1 above)
Comment (JeffreyThiessen) [reply progress/upload_status.py]: Fixed in effa23ba966ebf6801960a9a3075812ff0446a91

File: progress/upload_status.py (hunk 2 above)
Comment (JeffreyThiessen) [reply progress/upload_status.py]: Fixed in effa23ba966ebf6801960a9a3075812ff0446a91

File: tests_integration/test_end_to_end.py (hunk 3 above)
Comment (JeffreyThiessen) [reply tests_integration/test_end_to_end.py]: Fixed in effa23ba966ebf6801960a9a3075812ff0446a91

File: tests_integration/test_end_to_end.py (hunk 4 above)
Comment (JeffreyThiessen) [reply tests_integration/test_end_to_end.py]: Fixed in effa23ba966ebf6801960a9a3075812ff0446a91

File: tests_integration/test_end_to_end.py (hunk 5 above)
Comment (JeffreyThiessen) [reply tests_integration/test_end_to_end.py]: Fixed in effa23ba966ebf6801960a9a3075812ff0446a91
